dialogue-core/src/main/java/com/palantir/dialogue/core/BalancedNodeSelectionStrategyChannel.java

```diff
@@ -56,18 +55,22 @@
     private static final int UNHEALTHY_SCORE_MULTIPLIER = 2;
 
     private final BalancedScoreTracker tracker;
-    private final ImmutableList<LimitedChannel> channels;
+    private final HostAndLimitedChannels channels;
 
     BalancedNodeSelectionStrategyChannel(
-            ImmutableList<LimitedChannel> channels,
+            HostAndLimitedChannels channels,
```

Comment:
Not a suggestion for this change, just musing: Part of me wants to apply BalancedScoreTracker into HostAndLimitedChannels to consolidate our metrics. The data is useful for pin-until-error and single-host configurations, and it would open the door for reloads/strategy changes to retain existing knowledge of dead hosts.

Reply:
How would you use balancedscore tracker in pin until error? surely scores will be empty for the unpinned hosts?

Reply:
It depends -- if nodes fail frequently, we may rotate across nodes quickly. The score just isn't used by the pin-until-error strategy.

Reply:
> The score just isn't used by the pin-until-error strategy.

I know, but how would you propose it to be used?

Reply:
Dashboards for investigation. Unclear how useful that would be.